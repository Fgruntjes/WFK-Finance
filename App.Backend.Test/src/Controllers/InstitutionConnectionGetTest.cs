using App.Lib.Data.Entity;

namespace App.Backend.Test.Controllers;

public class InstitutionConnectionGetTest : IClassFixture<InstitutionConnectionFixture>
{
    private readonly InstitutionConnectionFixture _fixture;

    public InstitutionConnectionGetTest(InstitutionConnectionFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task Success()
    {
        // Act
        var result = await _fixture.Client.ExecuteQuery(new { _fixture.InstitutionConnectionEntity.Id });

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task OrganisationMismatch()
    {
        // Arrange
        var connectionEntity = new InstitutionConnectionEntity()
        {
            ExternalId = $"SomeExternalId-organisation-missmatch-0",
            ConnectUrl = new Uri($"https://www.example-organisation-missmatch-0.com/"),
            InstitutionId = _fixture.InstitutionEntity.Id,
            OrganisationId = _fixture.AltOrganisationId,
        };
        _fixture.SeedData(c =>
        {
            c.InstitutionConnections.Add(connectionEntity);
        });

        // Act
        var result = await _fixture.Client.ExecuteQuery(new { connectionEntity.Id });

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task WithInstitution()
    {
        // Act
        var result = await _fixture.Client.ExecuteQuery(new { _fixture.InstitutionConnectionEntity.Id });

        // Assert
        result.MatchSnapshot();
    }

    [Fact]
    public async Task WithAccounts()
    {
        // Arrange
        var InstitutionConnectionAccounts = new List<InstitutionConnectionAccountEntity>();
        for (int i = 0; i < 3; i++)
        {
            InstitutionConnectionAccounts.Add(new InstitutionConnectionAccountEntity()
            {
                ExternalId = $"SomeExternalId-organisation-account-{i}",
                InstitutionConnectionId = _fixture.InstitutionConnectionEntity.Id,
                Iban = $"IBAN{i}",
            });
        }
        _fixture.SeedData(c =>
        {
            c.InstitutionConnectionAccounts.AddRange(InstitutionConnectionAccounts);
        });

        // Act
        var result = await _fixture.Client.ExecuteQuery(new { Id = _fixture.InstitutionConnectionEntity.Id });

        // Assert
        result.MatchSnapshot();
    }
}