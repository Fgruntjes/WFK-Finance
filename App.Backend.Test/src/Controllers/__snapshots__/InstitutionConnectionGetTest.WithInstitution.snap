{
  "data": {
    "institutionConnection": {
      "get": {
        "externalId": "SomeExternalId-organisation-match-0",
        "institution": {
          "name": "Some Institution Name"
        }
      }
    }
  }
}
